### PR TITLE
feat: sticky header and changeable Sign In link

### PR DESCRIPTION
**Pull request by ikrestov**
This adds additional top level header link property className.

stoplight-menu

Sign In/Sign Up link text change is a impure hack in Header component. 
IMO, there is no clean way to drive it from configuration.

\cc @moaiandin

### CODE REVIEW

File: src/components/Header/config.js [hunk 1]
```diff
@@ -38,6 +38,12 @@ export default {
           type: 'array',
           required: false,
         },
+        {
```
Comment (lottamus) [inline src/components/Header/config.js]: I don't think we need to add this to the config? Why not just add the classname directly in the code?

File: netlify/settings.yaml [hunk 2]
```diff
@@ -115,6 +115,7 @@ header:
         - arrow-right
       isButton: true
       title: Sign In
+      className: sign-in-link
```
Comment (lottamus) [inline netlify/settings.yaml]: Just add this in the code directly

**Comment by ikrestov** [comment]
I have removed extra config entry and changed background to inherit from header (hero), or default to black.

Not sure why netlify build failed, it build fine locally.

File: src/components/Header/index.tsx [hunk 3]
```diff
@@ -212,29 +214,79 @@ export interface IHeader {
...
+
+  onUnfix() {
+    this.setState({unpinned: false});
+  }
+
+  public render() {
+    const { header, meta, color } = this.props;
+    const { unpinned } = this.state;
+    const bgClassName = unpinned ? `bg-${color || 'black'}` : '';
+    let headerItems = unpinned ? header.items.slice() : header.items;
```
Comment (lottamus) [inline src/components/Header/index.tsx]: Looks like its failing here. Just need to safeguard header && header.items before slicing

image

Comment (ikrestov) [reply src/components/Header/index.tsx]: Thanks. All green now.

**Comment by lottamus** [comment]
Great work on this! I think it looks awesome!

**Pull request merged**